server/public/model/config.go

```diff
@@ -4178,6 +4178,10 @@ func (s *TeamSettings) isValid() *AppError {
 		return NewAppError("Config.IsValid", "model.config.is_valid.sitename_length.app_error", map[string]any{"MaxLength": SitenameMaxLength}, "", http.StatusBadRequest)
 	}
 
+	if !*s.ExperimentalViewArchivedChannels {
```

Comment:
We should also update the definition of this field to avoid writing the config to json ever again and mark it as deprecated. Similar to how we did for gossip encryption:

```
// Deprecated: use EnableGossipEncryption
EnableExperimentalGossipEncryption *bool `json:",omitempty"`
```

And also remove this code:

```
if s.ExperimentalViewArchivedChannels == nil {
	s.ExperimentalViewArchivedChannels = NewPointer(true)
}
```